packages/actor-hash-quads-sha1/lib/ActorHashQuadsSha1.ts

```diff
@@ -0,0 +1,28 @@
+import type { IActionHashQuads, IActorHashQuadsOutput } from '@comunica/bus-hash-quads';
+import { ActorHashQuads } from '@comunica/bus-hash-quads';
+import type { IActorTest } from '@comunica/core';
+import { sha1 } from 'hash.js';
+
+// eslint-disable-next-line ts/no-require-imports,ts/no-var-requires
+const canonicalize = require('canonicalize');
+
+/**
+ * A comunica Memento Http Actor.
+ */
+export class ActorHashQuadsSha1 extends ActorHashQuads {
+  public async test(action: IActionHashQuads): Promise<IActorTest> {
+    if (!action.allowHashCollisions) {
+      throw new Error(`Actor ${this.name} can not provide hash functions without hash collisions`);
+    }
+    return true;
+  }
+
+  public async run(_action: IActionHashQuads): Promise<IActorHashQuadsOutput> {
+    return {
+      hashFunction: quads => sha1()
+        .update(canonicalize(Object.entries(quads)))
```

Comment:
This will not always work correctly if quads come from different data factories.
I would first convert each quad to its canonical string representation using for example `rdf-string`.

Reply:
I thought the `canonicalize` function might do that.
There is the `termToString` function which results in something like `<<https://lab.coret.org/id/comunica_testcase_1 http://schema.org/distribution https://lab.coret.org/id/comunica_testcase_1a>>` which I think might work. Is that a reliable method?

Reply:
Use quadToString

Reply:
Sorry, I still didn't not find a `quadToString` in `rdf-string`. There is `quadToStringQuad` but it returns an object.
I can do `JSON.stringify(quadToStringQuad(quad))`.

Or I could make a `quadToString` like [in MemoryPhysicalQueryPlanLogger.ts](https://github.com/comunica/comunica/blob/87baf2afed021a254859e64b92f34d9b51c6a7db/packages/actor-query-process-explain-physical/lib/MemoryPhysicalQueryPlanLogger.ts#L182)

Reply:
> JSON.stringify(quadToStringQuad(quad))

Yes, that's perfect!

Reply:
Great! Already implemented